zsh/prompt.zsh

```diff
@@ -0,0 +1,91 @@
+# autoload docs:
+# http://zsh.sourceforge.net/Doc/Release/Shell-Builtin-Commands.html colors
+# provides `$fg_bold` etc:
+# http://zsh.sourceforge.net/Doc/Release/User-Contributions.html#Other-Functions
+autoload -U colors && colors
+
+# Expand functions in the prompt
+setopt prompt_subst
+
+# Hey, colors
+prompt_color() {
+  [[ -n "$1" ]] && print "%{$fg_bold[$2]%}$1%{$reset_color%}"
+}
+
+prompt_gray()   { print "$(prompt_color "$1" grey)"  }
+prompt_yellow() { print "$(prompt_color "$1" yellow)"  }
+prompt_green()  { print "$(prompt_color "$1" green)"  }
+prompt_red()    { print "$(prompt_color "$1" red)"  }
+prompt_cyan()   { print "$(prompt_color "$1" cyan)"  }
+prompt_blue()   { print "$(prompt_color "$1" blue)"  }
+prompt_magenta(){ print "$(prompt_color "$1" magenta)"  }
+
+# If passing this a non-empty string, it will print prefix with a space.
+prompt_spaced() { [[ -n "$1" ]] && print " $@" }
+
+# Adds the current branch name
+git_prompt_info() {
+  ref=$(git symbolic-ref HEAD 2> /dev/null)
+
+  if [[ -n $ref ]]; then
```

Comment:
I'd quote `$ref`: `if [[ -n "$red" ]]`

More here, under "quote all the freaking time": https://robots.thoughtbot.com/the-unix-shells-humble-if#the--command